src/bin/tikv-ctl.rs

```diff
@@ -21,25 +21,381 @@ extern crate clap;
 extern crate protobuf;
 extern crate kvproto;
 extern crate rocksdb;
+extern crate grpcio;
+extern crate futures;
 #[cfg(test)]
 extern crate tempdir;
 extern crate rustc_serialize;
 
-use std::{str, u64};
-use clap::{App, Arg, SubCommand};
+use std::{process, str, u64};
+use std::error::Error;
+use std::sync::Arc;
+use std::path::PathBuf;
 use rustc_serialize::hex::{FromHex, ToHex};
+
+use clap::{App, Arg, ArgMatches, SubCommand};
 use protobuf::Message;
+use futures::{future, Future, Stream};
+use grpcio::{ChannelBuilder, Environment, Error as GrpcError};
+use protobuf::RepeatedField;
+use protobuf::text_format::print_to_string;
+
 use kvproto::raft_cmdpb::RaftCmdRequest;
-use kvproto::raft_serverpb::{PeerState, RaftApplyState, RaftLocalState, RegionLocalState};
-use kvproto::eraftpb::Entry;
-use rocksdb::{ReadOptions, SeekKey, DB};
+use kvproto::raft_serverpb::PeerState;
+use kvproto::eraftpb;
+use kvproto::kvrpcpb::MvccInfo;
+use kvproto::debugpb::*;
+use kvproto::debugpb::DB as DBType;
+use kvproto::debugpb_grpc::DebugClient;
 use tikv::util::{self, escape, unescape};
-use tikv::util::codec::bytes::encode_bytes;
-use tikv::raftstore::store::keys;
-use tikv::raftstore::store::engine::{IterOption, Iterable, Peekable};
-use tikv::storage::{CfName, ALL_CFS, CF_DEFAULT, CF_LOCK, CF_RAFT, CF_WRITE};
-use tikv::storage::mvcc::{Lock, Write};
-use tikv::storage::types::Key;
+use tikv::raftstore::store::{keys, Engines};
+use tikv::raftstore::store::debug::{Debugger, RegionInfo};
+use tikv::storage::{ALL_CFS, CF_DEFAULT, CF_LOCK, CF_RAFT, CF_WRITE};
+
+enum DebugExecutor {
```

Comment:
is it better to define an Executor trait so we can avoid following `match` in many functions?